go/arrow/numeric.tmpldata

```diff
@@ -107,5 +107,29 @@
     "Opt": {
       "Parametric": true
     }
+  },
+  {
+    "Name": "Date32",
+    "name": "date32",
+    "Type": "Date32",
+    "QualifiedType": "arrow.Date32",
+    "InternalType": "int32",
+    "Default": "0",
+    "Size": "4",
+    "Opt": {
+      "Parametric": true
```

Comment:
AFAIK, they're not parametric.

Reply:
Makes sense, there's only a single Date32 unit (same as Date64).

However, when I remove this, the generated code in `numericbuilder.gen.go` uses `arrow.PrimitiveTypes.Date32` within the function `Date32Builder.newData`, but Date32 isn't added to PrimitiveTypes in `datatype_numeric.gen.go`. Is this a bug in the generator or is there something else I need to add? Are both "QualifiedType" and "InternalType" meant to stay?